src/Exception/AuthenticationException.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Joindin\Api\Exception;
+
+use \Exception;
+use \Throwable;
+
+final class AuthenticationException extends Exception implements Throwable
+{
+    private const MESSAGE = 'You must be logged in to perform this operation.';
+
+    public static function forUnauthenticatedUser(string $message = null): self
+    {
+        return new static($message ?? self::MESSAGE, 401);
```

Comment:
What do you think about using the corresponding constant from `teapot/status-code`?

```diff
diff --git a/src/Exception/AuthenticationException.php b/src/Exception/AuthenticationException.php
index 4becf9f..2635e47 100644
--- a/src/Exception/AuthenticationException.php
+++ b/src/Exception/AuthenticationException.php
@@ -5,6 +5,7 @@ declare(strict_types=1);
 namespace Joindin\Api\Exception;

 use \Exception;
+use Teapot\StatusCode\Http;
 use \Throwable;

 final class AuthenticationException extends Exception implements Throwable
@@ -13,6 +14,6 @@ final class AuthenticationException extends Exception implements Throwable

     public static function forUnauthenticatedUser(string $message = null): self
     {
-        return new static($message ?? self::MESSAGE, 401);
+        return new static($message ?? self::MESSAGE, Http::UNAUTHORIZED);
     }
 }
```